spec/acceptance/helpers/system_gem_helper.rb

```diff
@@ -1,25 +1,33 @@
+# frozen_string_literal: true
+
 module Acceptance
   module Helpers
     module SystemGemHelper
-
       def install_system_factor_gem(host)
         host.install_package('rubygems')
         # Facter 4.0 needs ruby 2.3 or later and system ruby on EL7
         # is version 2.0.0
-        on(host, '/usr/bin/gem install facter --version "<4.0.0"')
+        on(host, '/usr/bin/gem install facter || /usr/bin/gem install facter --version "<4.0.0"')
+
+        # Facter 2 hangs on EC2 fact if not on a cloud system
+        result = on(host, 'timeout 5 facter networking', :accept_all_exit_codes => true)
+        unless result.exit_code == 0
+          # The EC2 fact is hanging due to a bug in facter 2, kill it with fire
+          on(host, %{find $( gem env gemdir ) -name ec2.rb | grep -e facter | xargs -r rm})
+        end
 
         # beaker-helper fact_on() now uses '--json' on facter calls, so
         # we need to make sure the json gem is installed
         result = on(host, 'facter --json fqdn', :accept_all_exit_codes => true)
-        if result.exit_code != 0
-          # We have old system Ruby (1.8.7) which does not include json.  So,
-          # install the pure-Ruby version of json.
-          on(host, "gem install json_pure --version '<2.0.0'")
-        end
+        return if result.exit_code.eql?(0)
+
+        # We have old system Ruby (1.8.7) which does not include json.  So,
```

Comment:
The last OS with a system Ruby of <2.0.0 was EL6 (which we no longer support) and AFAICT all EL6 SUTs in our acceptance tests have been removed.

Would it be better to just remove this code entirely?